.github/workflows/ci.yml

```diff
@@ -73,32 +33,76 @@ jobs:
           - omnibus
           - python
           - terraform
+    env:
+      DOCKER_BUILDKIT: 1
+      DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+      DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
     steps:
-      - name: Pull image from packages
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Prepare environment variables
         run: |
-          docker login docker.pkg.github.com -u x -p ${{ secrets.GITHUB_TOKEN }}
-          docker pull "docker.pkg.github.com/${{ github.repository }}/dependabot-core-ci:${{ github.sha }}"
-          docker logout
-      - name: Run Rubocop linting
+          BRANCH_REF="$(echo '${{ github.ref }}' | sed -E 's/[^A-Za-z0-9]+/-/g')"
+          echo ::set-env name=BRANCH_REF::$(BRANCH_REF)
+      - name: Log in to the Docker registry
         run: |
-          docker run --rm docker.pkg.github.com/${{ github.repository }}/dependabot-core-ci:${{ github.sha }} bash -c "cd /home/dependabot/dependabot-core/${{ matrix.suite }} && bundle exec rubocop . -c ../.rubocop.yml"
-      - name: Run ${{ matrix.suite }} tests with rspec
+          if [ -n "$DOCKER_USERNAME" ] && [ -n "$DOCKER_PASSWORD" ]; then
```

Comment:
This could maybe be slicker as an Actions-native conditional, but I'm not sure of the syntax...

https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions says `secrets` is in scope - maybe `if: ${{secrets. DOCKER_USERNAME && secrets. DOCKER_PASSWORD}}` ?

The only hit I could find is: https://github.com/Kegbot/kegbot-server/blob/54b02880e8b440168110985230a60b7390e10991/.github/workflows/docker.yml#L41 :

`if: ${{ '${{ secrets.DOCKER_HUB_PASSWORD }}' != '' && '${{ secrets.DOCKER_HUB_USERNAME }}' != '' }}` .

Reply:
Idea: drop a breadcrumb like `echo ::set-env name=DOCKER_LOGGED_IN::true` for later steps to use, instead of duplicating this.

Reply:
> https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions says `secrets` is in scope - maybe `if: ${{secrets. DOCKER_USERNAME && secrets. DOCKER_PASSWORD}}` ?

Yeah tried that but got a syntax error saying it wasn't available 😕